inc/task.class.php

```diff
@@ -923,80 +936,98 @@ function forceRunning() {
+      if (isset($_SESSION['glpiactiveentities_string'])) {
+         $entity_restrict_task = getEntitiesRestrictRequest("AND", 'task');
       }
 
-      $query_select = [];
-      foreach ($query_fields as $index => $key) {
-         $query_select[] = $key[1] . " AS '" . $key[0] . "'";
-      }
+      PluginFusioninventoryToolbox::logIfExtradebug(
+         "pluginFusioninventory-tasks",
+         "Get tasks jobs log, tasks list: ". implode(",", $task_ids));
 
-      $query_joins         = [];
-      $query_joins['task'] = "
-         INNER JOIN `glpi_plugin_fusioninventory_tasks` as task
-            ON job.`plugin_fusioninventory_tasks_id` = task.`id`
-            AND task.`is_active` = 1";
+      $prepare_chrono = [
+         "start" => microtime(true),
+         "end"   => 0
+      ];
 
+      // We get list of taskjobs
+      $active_task = $only_active ? "AND task.`is_active`='1'" : "";
       $data_structure = [
          'query' => "SELECT
-                    ".implode(",\n", $query_select)."
-                    FROM `glpi_plugin_fusioninventory_taskjobs` as job
-                    ".implode("\n", $query_joins)."
-                    ".implode("\n", $query_where),
-         'result' => null
+            `job`.`id` as 'job.id',
+            `job`.`name` as 'job.name',
+            `job`.`method` as 'job.method',
+            `job`.`targets` as 'job.targets',
+            `task`.`id` as 'task.id',
+            `task`.`name` as 'task.name'
+            FROM `glpi_plugin_fusioninventory_taskjobs` as job
+            LEFT JOIN `glpi_plugin_fusioninventory_tasks` as task
+              ON job.`plugin_fusioninventory_tasks_id` = task.`id` $active_task $tasks_list $entity_restrict_task
+            WHERE task.`id` IS NOT NULL ",
+         'result' => null,
+         "start" => microtime(true),
+         "end"   => 0
       ];
 
       $data_structure['result'] = $DB->query($data_structure['query']);
+      PluginFusioninventoryToolbox::logIfExtradebug(
+         "pluginFusioninventory-tasks",
+         "Preparing query: " . print_r($data_structure, true));
+      if ($data_structure['result']->num_rows <= 0) {
+         // Not useful to go further, we will not have any result to send!
+         // Perharps the required tasks are not even active ;)
+         return ['tasks' => $logs, 'agents' => $agents];
+      }
 
-      //Results grouped by tasks > jobs > jobstates
-      $logs          = [];
-
-      //Target cache (used to speed up data formatting)
-      $targets_cache = [];
+      // Target cache (used to speed up data formatting)
       $expanded      = [];
       if (isset($_SESSION['plugin_fusioninventory_tasks_expanded'])) {
          $expanded = $_SESSION['plugin_fusioninventory_tasks_expanded'];
       }
 
-      while ($result = $data_structure['result']->fetch_row()) {
-         $task_id = $result[$fieldmap['task.id']];
+      $agent_state_types = [
+         'agents_prepared', 'agents_running', 'agents_cancelled',
```

Comment:
Agent states are used several times in the code. Please consider adding class constants